java-frontend/src/main/java/org/sonar/java/SonarComponents.java

```diff
@@ -305,6 +307,16 @@ public void registerTestSharedCheck(JavaCheck check, Collection<RuleKey> ruleKey
     }
   }
 
+  @Override
+  public void registerCustomFileScanner(RuleScope ruleScope, JavaFileScanner scanner) {
+    if (ruleScope != RuleScope.TEST) {
+      mainChecks.add(scanner);
+    }
+    if (ruleScope != RuleScope.MAIN) {
+      testChecks.add(scanner);
+    }
```

Comment:
Can we arrange these tests in a less clever but more explicit way?
```suggestion
    if (ruleScope == RuleScope.MAIN || ruleScope ==  RuleScope.ALL) {
      mainChecks.add(scanner);
    }
    if (ruleScope == RuleScope.TEST || ruleScope ==  RuleScope.ALL) {
      testChecks.add(scanner);
    }
```

Reply:
Fixed.